src/pymmcore_widgets/_pixel_size_widget.py

```diff
@@ -0,0 +1,590 @@
+        if not self._objective_device:
+            return
+
+        self._magnification.clear()
+
+        obj_list = self._mmc.getStateLabels(self._objective_device)
+        self.table.setRowCount(len(obj_list))
+
+        obj_cfg_px = self._get_px_info()
+
+        cam_px_guess = 0.0
+
+        for idx, objective in enumerate(obj_list):
+            objective_item = QtW.QTableWidgetItem(objective)
+            objective_item.setFlags(Qt.ItemIsEnabled | Qt.ItemIsSelectable)
+            self.table.setItem(idx, OBJECTIVE_LABEL, objective_item)
+
+            cfg_name = "None"
+            px_value = 0.0
+            total_mag = 0.0
+
+            for obj, cfg, px in obj_cfg_px:
+                if objective == obj:
+                    cfg_name = cfg
+                    px_value = px
+                    break
+
+            self._set_item_in_table(idx, RESOLUTION_ID, cfg_name)
+            red_id_item = self.table.cellWidget(idx, RESOLUTION_ID)
+            red_id_item.setProperty("resID", red_id_item.text())
+
+            self._set_item_in_table(idx, IMAGE_PX_SIZE, f"{px_value:.4f}")
+
+            mag_value = (
+                int(match.groups()[0])
+                if (match := re.search(r"(\d{1,3})[xX]", objective))
+                else 0.0
+            )
+            total_mag = mag_value * self._mmc.getMagnificationFactor()
+
+            self._set_item_in_table(idx, MAGNIFICATION, total_mag)
+            self._magnification.append([objective, total_mag])
+
+            self._set_item_in_table(idx, CAMERA_PX_SIZE, 0.0)
+
+            if total_mag != 0.0 and px_value != 0.0 and cfg_name:
+                cam_px_guess = total_mag * px_value
+
+            self._add_delete_btn(idx)
+            self._update_status(idx)
+
+        for r in range(self.table.rowCount()):
+            cam_px = self.table.cellWidget(r, CAMERA_PX_SIZE)
+            with signals_blocked(cam_px):
+                cam_px.setText(f"{cam_px_guess:.2f}")
+
+        self._update_wdg_size()
+
+    def _get_px_info(self) -> List[Tuple[str, str, float]]:
+        obj_cfg_px = []
+        for cfg in self._mmc.getAvailablePixelSizeConfigs():
+            obj = list(self._mmc.getPixelSizeConfigData(cfg))[0][2]
+            px_value = self._mmc.getPixelSizeUmByID(cfg)
+            obj_cfg_px.append((obj, cfg, px_value))
+        return obj_cfg_px
+
+    def _set_item_in_table(self, row: int, col: int, value: Any) -> None:
+        item = QtW.QLineEdit(text=str(value))
+        item.setProperty("row", row)
+        item.setProperty("col", col)
+        item.setAlignment(Qt.AlignCenter)
+        item.setFrame(False)
+        if self._is_read_only(col):
+            item.setReadOnly(True)
+            item.setStyleSheet("color:magenta")
+        else:
+            item.setReadOnly(False)
+            item.setStyleSheet("")
+
+        if col in {2, 3, 4}:
+            regex = QRegularExpression(FLOAT_REGEX)
```

Comment:
there is a QDoubleValidator made just for this: 
https://doc.qt.io/qt-6/qdoublevalidator.html